src/frontend/src/eth/services/eth-transactions.services.ts

```diff
@@ -113,29 +114,43 @@ const loadErc20Transactions = async ({
 		return { success: false };
 	}
 
-	try {
-		const { transactions: transactionsRest } = etherscanRests(networkId);
-		const transactions = await transactionsRest({ contract: token, address });
-		ethTransactionsStore.set({ tokenId, transactions });
-	} catch (err: unknown) {
-		ethTransactionsStore.reset();
-
-		const {
-			transactions: {
-				error: { loading_transactions_symbol }
-			}
-		} = get(i18n);
-
-		toastsError({
-			msg: {
-				text: replacePlaceholders(loading_transactions_symbol, {
-					$symbol: token.symbol
-				})
-			},
-			err
-		});
+	const result = await restRequest({
+		request: async () => {
+			const { transactions: transactionsRest } = etherscanRests(networkId);
+			return await transactionsRest({ contract: token, address });
+		},
+		onSuccess: (transactions) => {
+			ethTransactionsStore.set({ tokenId, transactions });
+			return { success: true };
+		},
+		onError: (err) => {
+			ethTransactionsStore.reset();
+
+			const {
+				transactions: {
+					error: { loading_transactions_symbol }
+				}
+			} = get(i18n);
+
+			toastsError({
+				msg: {
+					text: replacePlaceholders(loading_transactions_symbol, {
+						$symbol: token.symbol
+					})
+				},
+				err
+			});
+
+			return { success: false };
+		},
+		onRetry: async () => {
+			await new Promise((resolve) => setTimeout(resolve, Math.floor(Math.random() * 1000 + 1000)));
+		}
+	});
+
+	if (isNullish(result)) {
```

Comment:
since I put `onError` as optional in `restRequest`, it may happen that on errors it return nothing (`undefined`); but even if `onError` is given, I put that the return can either be the same as `onSuccess` or `undefined` (in case of function `() => void` )

shall I remove all these liberties? for example `onError` MUST be given and MUST return the same as `onSuccess`?

Reply:
Thanks for the explanation, now I understand, it makes sense the result can be nullish. 

Now, what are you returning `return { success: false };` if `result` is `undefined`?

Reply:
Just a guard kind of do I get it right?

Reply:
Or should actually `restRequest` returns `Promise<void>`, that way no need of a guard?